src/store.ts

```diff
@@ -102,22 +111,54 @@ export class Store<T> {
     this._state.next(state);
   }
 
-  public dispatch<P extends any[]>(reducer: Reducer<T, P> | string, ...params: P) {
-    let action: Reducer<T, P>;
+  public dispatch<P extends any[]>(reducer: Reducer<T, P> | string, ...params: P): Promise<void> {
+    const action = this.lookupAction(reducer as Reducer<T> | string);
+    if (!action) {
+      return Promise.reject(new Error(
+        `Tried to dispatch an unregistered action ${reducer && (typeof reducer === "string" ? reducer : reducer.name)}`

```

Comment:
I've noticed that we're repeating the same error three times during the dispatch cycle. This should be extracted into a single place

Reply:
Good idea, I have extracted this error message into a new Error class UnregisteredActionError. I hope you imagined something along the lines of that